R/build_lookup.R

```diff
@@ -89,32 +89,43 @@ pad_single_digit <- function(gene_str) {
   return(gsub("([A-Za-z]+)(\\d)([-\\*])", "\\10\\2\\3", gene_str))
 }
 
-#' Create lookup tables from reference FASTAs
+#' Create these lookup tables from a directory of FASTA files:
 #'
-#' Create lookup tables within in a given directory that contains FASTA files:
 #'    - lookup.csv
 #'    - lookup_from_tenx.csv
 #'    - lookup_from_adaptive.csv
 #'
+#' The lookup tables are stored in an application data folder via rappdirs. For example:
+#'
+#'    - MacOS: ``~/Library/Application Support/<AppName>``
+#'    - Windows: ``C:\Documents and Settings\<User>\Application Data\Local Settings\<AppAuthor>\<AppName>``
+#'    - Linux: ``~/.local/share/<AppName>``
+#'
 #' @param data_dir A string, the directory containing FASTA files.
+#' @param species A string, the name of species that will be used when running TCRconvert with these lookup tables.
 #'
 #' @return Nothing.
 #' @autoglobal
 #' @export
 #' @examples
 #' # For the example, create and use a temporary folder
-#' fastadir <- file.path(tempdir(), "tcrconvertr_tmp")
+#' fastadir <- file.path(tempdir(), "TCRconvertR_tmp")
 #' dir.create(fastadir)
 #' trav <- get_example_path("fasta_dir/test_trav.fa")
 #' trbv <- get_example_path("fasta_dir/test_trbv.fa")
 #' file.copy(c(trav, trbv), fastadir)
 #'
 #' # Build lookup tables
-#' build_lookup_from_fastas(fastadir)
+#' build_lookup_from_fastas(fastadir, "rabbit")
 #'
```

Comment:
i'm curious what happens if the user passes in a string with spaces in it? Or other symbols like `.` or `-`, etc.? Maybe it's fine but I'd suggest thinking about this. You could make sure they can only pass in a certain type of string, or don't check the input (and you may not need to, I just dont know).

Reply:
You're totally right - this string becomes a folder name so it's important that it's valid. I added a check for forbidden characters and a corresponding test in 5f4ebae0cdb801642f854a97546410fd30e18f6a.

Reply:
looks good